src/Illuminate/Foundation/Application.php

```diff
@@ -530,6 +536,15 @@ protected function markAsRegistered($provider)
 		$this->serviceProviders[] = $provider;
 
 		$this->loadedProviders[$class] = true;
+		
+		$aliases = class_parents($class) + class_implements($class) + array($class);
+		foreach ($aliases as $alias)
+		{
+			if ( ! isset($this->registeredProviders[$alias]))
+			{
+				$this->registeredProviders[$alias] = $provider;
```

Comment:
I did a quick microbenchmark, and it looks like `+= array_fill_keys($aliases, $provider);` performs worse than `foreach` / `!isset` / `=` for a small `$aliases` array.  But not significantly, so it probably isn't important either way.